runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -11551,9 +11551,9 @@ VMgenerateCatchBlockBBStartPrologue(
    TR::Compilation *comp = cg->comp();
    TR_J9VMBase *fej9 = (TR_J9VMBase *)(cg->fe());
 
-   if (comp->getOption(TR_FullSpeedDebug))
+   if (comp->getOption(TR_FullSpeedDebug) || comp->getOptions()->isHookEnabled(TR::Options::J9HookExceptionCatch))
```

Comment:
Does FSD imply that we should always configure the JIT method bodies for all the hooks? In which case we should be checking only the `TR::Options::J9HookExceptionCatch` here. Where FSD is enabled we should enabled all the corresponding hooks that we need.

Reply:
Good suggestion. It has been updated accordingly.